src/build-scripts/hydrafw-version.py

```diff
@@ -14,7 +16,12 @@
 if len(args)==1:
   sys.stdout = open(args[0], 'w')
   git=Repo(search_parent_directories=True).git
-  print('#define HYDRAFW_GIT_TAG "' + git.describe(tags=True,always=True,dirty=True,long=True) + '"')
+  vermagic=git.describe(tags=True,always=True,dirty=True,long=True)
+  if os.environ.get("GITHUB_CI_PR_SHA", "") != "":
+    sha_id=os.environ["GITHUB_CI_PR_SHA"][:7]
+    print('#define HYDRAFW_GIT_TAG "' + re.sub('-g.*$', '-g' + sha_id, vermagic) + '"')
+  else:
+    print('#define HYDRAFW_GIT_TAG "' + vermagic + '"')
```

Comment:
I know that on the long run the goal is to remove python dependency at all and not to extend it even deeper, but currently I have to work with what we have. Which is adopting of extraction proper git tag at the moment of build on GitHub server. Although after helpful conversations & recommendations I did manage to push through all necessary git meta information for the build, one essential GitHub build issue still remains: the repo is in detached state at the time of build, so we can't get valid the most recent committed sha id easily. Therefore I had to use already tested approach: we set env variable in `push.yml` with actual sha id since GitHub allows to extract this information at the time of build. And then during build we check if sha id env is set, and then we replace detached sha id by actual one. I guess it's self-explanatory anyway but just for those who may be curious.

Reply:
All seems good when checking the latest hydrafw.dfu
Do you plan to do some cleanup for example in scripts/env.sh ... (before to merge it) ?

Reply:
> All seems good when checking the latest hydrafw.dfu

Perfecto!

> Do you plan to do some cleanup for example in scripts/env.sh ... (before to merge it) > ?

Yes-yes, sure, of course! Just give me a day or so since I plan to make a couple more commits here, and then I will tell you when it's ready. Thanks.

Reply:
Oookay, I guess now it should be ok. So consider it done on my side for now. But if you have any other additional suggestions or any feedback, just let me know.

P.S. I'm sincerely sorry if some of my questions or explanations were a bit intimidating or irritating here in the conversations ... but I was just trying to be straight-forward and direct only to make sure that I understand clearly what exactly must be implemented for this PR since build scripts & steps for `hydrafw` in the context of GitHub Actions environment turned out to be a little bit more tricky than I thought at first look.